src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java

```diff
@@ -279,58 +284,47 @@ public int allowedSpaceCount()
-            if (1 == currentWordIndex)
+            final String[] citizenNameParts = citizenName.split(" ");
+            if (citizenNameParts.length > currentWordIndex)
             {
-                final Set<String> middleNameSet = new HashSet<>();
-                for (final String citizenName : citizenNameStrings)
+                int i = 0;
+                while (i < currentWordIndex)
                 {
-                    final String[] citizenNameParts = citizenName.split(" ");
-                    if (potentiaCitizenNameParts[0].equals(citizenNameParts[0]))
+                    if (!potentiaCitizenNameParts[i].equals(citizenNameParts[i]))
                     {
-                        middleNameSet.add(citizenNameParts[1]);
+                        break;
                     }
+                    ++i;
                 }
-                return middleNameSet.stream().filter(k -> k.startsWith(potentiaCitizenNameParts[1])).collect(Collectors.toList());
-            }
-            else
-            {
-                final Set<String> lastNameSet = new HashSet<>();
-                for (final String citizenName : citizenNameStrings)
+                if (i == currentWordIndex)
                 {
-                    final String[] citizenNameParts = citizenName.split(" ");
-                    if ((potentiaCitizenNameParts[0].equals(citizenNameParts[0])) && (potentiaCitizenNameParts[1].equals(citizenNameParts[1])))
-                    {
-                        lastNameSet.add(citizenNameParts[2]);
-                    }
+                    nameAtIndexStringSet.add(citizenNameParts[currentWordIndex]);
                 }
-                return lastNameSet.stream().filter(k -> k.startsWith(potentiaCitizenNameParts[2])).collect(Collectors.toList());
-           }
+            }
+        }
+        if (!nameAtIndexStringSet.isEmpty())
+        {
+            return nameAtIndexStringSet.stream().filter(k -> k.startsWith(potentiaCitizenNameParts[currentWordIndex])).collect(Collectors.toList());
+        }
+        else
+        {
+            return Collections.emptyList();
         }
-        return Collections.emptyList();
     }
 
     @Nullable
-    public Object parse(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @Nullable final BlockPos pos,
-            @NotNull final List<ActionMenuHolder> parsedHolders,
+    public Object parse(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @Nullable final BlockPos pos, 
+            @NotNull final ModuleContext moduleContext,
+            @NotNull final ActionMenuState actionMenuState,
             final String potentialArgumentValue)
     {
+        final MineColonyDataProvider mineColonyDataProvider = moduleContext.get(MineColonyDataProvider.class);
+
         // TODO: selector support, such as used by CommandKill to find player
         // Entity entity = <net.minecraft.command.CommandBase>.getEntity(server, sender, args[0]);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)